src/System.Data.SqlClient/tests/ManualTests/SQL/AsyncTest/ReadAsyncTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public class ReadAsyncTest
+    {
+        [CheckConnStrSetupFact]
+        public static void Run()
+        {
+            CompareSyncAndAsync();
+        }
+
+        public static async void CompareSyncAndAsync()
+        {
+            double elapsedAsync = await RunReadAsync();
+            double elapsedSync = RunRead();
+            Assert.True(elapsedAsync < (elapsedSync / 2.0d));
+        }
+
+        private const string commandText = "SET NOCOUNT ON"
+                                        + " SELECT 'a'"
+                                        + " DECLARE @t DATETIME = SYSDATETIME()"
+                                        + " WHILE DATEDIFF(s, @t, SYSDATETIME()) < 20 BEGIN"
+                                        + "   SELECT 2 x INTO #y"
+                                        + "   DROP TABLE #y"
+                                        + " END"
+                                        + " SELECT 'b'";
+
+        public static async Task<double> RunReadAsync()
+        {
+            double maxElapsedTimeMillisecond = 0;
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                connection.Open();
+                using (SqlCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = commandText;
+                    using (SqlDataReader reader = command.ExecuteReader())
```

Comment:
You could use the async equivalent of ExecuteReader to make this method async. The same applies to connection.Open()